go/libraries/doltcore/sqle/dtables/ci_workflows.go

```diff
@@ -0,0 +1,432 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dtables
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+	"github.com/dolthub/vitess/go/sqltypes"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema/typeinfo"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dsess"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/index"
+	"github.com/dolthub/dolt/go/store/hash"
+	"github.com/dolthub/dolt/go/store/prolly"
+	stypes "github.com/dolthub/dolt/go/store/types"
+)
+
+const workflowsDefaultRowCount = 10
+
+var _ sql.StatisticsTable = (*WorkflowsTable)(nil)
+var _ sql.Table = (*WorkflowsTable)(nil)
+
+var _ sql.UpdatableTable = (*WorkflowsTable)(nil)
+
+var _ sql.DeletableTable = (*WorkflowsTable)(nil)
+var _ sql.InsertableTable = (*WorkflowsTable)(nil)
+
+var _ sql.ReplaceableTable = (*WorkflowsTable)(nil)
+
+var _ sql.IndexAddressableTable = (*WorkflowsTable)(nil)
```

Comment:
If you don't want to support indexes these tables, then you shouldn't need to implement this interface